TestUtilities/ProtobufComparison/src/main/java/com.coursesketch.test.utilities/ProtobufComparison.java

```diff
@@ -142,13 +155,17 @@ private void equals(final GeneratedMessage expected, final GeneratedMessage actu
      * @param incorrectFields A map containing fields that were found to be incorrect.
      */
     private void compareFields(final Object expectedValue, final Object actualValue, final Descriptors.FieldDescriptor field,
-            final Map<Descriptors.FieldDescriptor, ExpectationPair<Object, Object>> incorrectFields) {
+            final Map<Descriptors.FieldDescriptor, List<ExpectationPair<Object, Object>>> incorrectFields) {
+        if (field.isRepeated()) {
+            compareRepeatedFields(expectedValue, actualValue, field, incorrectFields);
+            return;
+        }
         if (expectedValue instanceof GeneratedMessage) {
             if (!(actualValue instanceof GeneratedMessage)) {
                 Assert.fail("Different types");
             }
             if (isDeepEquals) {
```

Comment:
I find that else if are messier than just an if.

Because it does stop that means it is the equivalent of an else if